Sources/RInAppMessaging/Views/FullView.swift

```diff
@@ -304,7 +307,7 @@ internal class FullView: UIView, FullViewType, RichContentBrowsable {
         carouselView.setPageControlVisibility(isHdden: layout != .carousel)
         buttonsContainer.isHidden = !viewModel.showButtons
         optOutView.isHidden = !viewModel.showOptOut
-        optOutButtonTopSpacer.isHidden = layout != .carousel && (!buttonsContainer.isHidden || !optOutView.isHidden)
+        optOutButtonTopSpacer.isHidden = layout == .carousel && (buttonsContainer.isHidden || optOutView.isHidden)
```

Comment:
what is the purpose of this property? what is the condition check for this?

Reply:
this is an extra spacing which i am hiding when there is carousel view , since the pageviewcontroller alraedy adds some spacing .